### ISSUE TITLE
manual bazel RBE builds are broken with bazel 0.20

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#17422]
Reproduction is to run bazel RBE build from your workstation:
```
$ bazel --bazelrc=tools/remote_build/manual.bazelrc build //test/...
Starting local Bazel server and connecting to it...
INFO: Invocation ID: d3599496-e498-4943-a6b3-e0c7807a0bf6
INFO: Streaming Build Event Protocol to https://source.cloud.google.com/results/invocations/d3599496-e498-4943-a6b3-e0c7807a0bf6
INFO: Repository rule 'io_bazel_rules_python' returned: {"remote": "https://github.com/bazelbuild/rules_python.git", "commit": "8b5d0683a7d878b28fffe464779c8a53659fc645", "shallow_since": "2018-04-16", "init_submodules": False, "verbose": False, "strip_prefix": "", "patches": [], "patch_tool": "patch", "patch_args": ["-p0"], "patch_cmds": [], "name": "io_bazel_rules_python"}
INFO: Repository rule 'org_pubref_rules_protobuf' returned: {"remote": "https://github.com/ghostwriternr/rules_protobuf", "commit": "1623a5a083f2a702bd62692d059353b4e06f3a5e", "shallow_since": "2018-07-30", "init_submodules": False, "verbose": False, "strip_prefix": "", "patches": [], "patch_tool": "patch", "patch_args": ["-p0"], "patch_cmds": [], "name": "org_pubref_rules_protobuf"}
ERROR: cc_toolchain_suite '@com_github_bazelbuild_bazeltoolchains//configs/ubuntu16_04_clang/1.1/bazel_0.16.1/default:toolchain' does not contain a toolchain for CPU 'k8', you may want to add an entry for 'k8|clang' into toolchains and toolchain_identifier 'linux_gnu_x86' into the corresponding cc_toolchain rule (see --incompatible_disable_cc_toolchain_label_from_crosstool_proto).
INFO: Elapsed time: 9.664s
INFO: 0 processes.
INFO: Build Event Protocol upload finished successfully
INFO: Build Event Protocol results available at https://source.cloud.google.com/results/invocations/d3599496-e498-4943-a6b3-e0c7807a0bf6
FAILED: Build did NOT complete successfully (52 packages loaded)
```

Workaround is to download an older version of bazel (our builds on kokoro are using on older bazel version as well, that's why the are working).
```
mkdir ~/bazel-0.17
cd ~/bazel-0.17
wget -q https://github.com/bazelbuild/bazel/releases/download/0.17.1/bazel-0.17.1-linux-x86_64
chmod u+x bazel-0.17.1-linux-x86_64
# use the 0.17 bazel binary for running RBE builds and things will work fine
```

**Comment by jtattermusch** [comment]
CC @vjpai

**Comment by jtattermusch** [comment]
We can consult with Foundry team.

**Comment by nicolasnoble** [comment]
No, it's not a blocker.

**Comment by jtattermusch** [comment]
I think the PR that @nicolasnoble  mentions here https://github.com/grpc/grpc/issues/17422#issuecomment-444934090 is https://github.com/grpc/grpc/pull/17385

**Comment by billfeng327** [comment]
Is it somehow related to this issue? [#6694](https://github.com/bazelbuild/bazel/issues/6694)

**Comment by nlopezgi** [comment]
These RBE builds are running using an old toolchain config that is not compatible with Bazel 0.20.0 (note the error points out that `@com_github_bazelbuild_bazeltoolchains//configs/ubuntu16_04_clang/1.1/**bazel_0.16.1**/default:toolchain` was used).
You need to update to the latest pin of the bazel-toolchains repo (https://releases.bazel.build/bazel-toolchains.html) and use the recommended toolchain configs for the version of Bazel you are using (if its 0.20.0, please use flags recommended in https://github.com/bazelbuild/bazel-toolchains/tree/master/configs/ubuntu16_04_clang/1.1/bazel_0.20.0)

**Comment by billfeng327** [comment]
[#17678](https://github.com/grpc/grpc/pull/17678) created, local bazel manual build did not prompt the same error but prompted some other unrelated error. I believe it resolved the toolchains correctly.

**Comment by jtattermusch** [comment]
Fixed by #17678 and #17684
